test/test_utils.py

```diff
@@ -47,3 +48,27 @@ def parse_common_options(datadir=None,
   sys.path.append(os.path.join(os.path.dirname(xla_folder), 'test'))
   sys.path.insert(0, xla_folder)
   return args
+
+
+def print_training_update(device_str, step_num, loss, rate, global_rate):
+  """Prints the training metrics at a given step.
+
+  Args:
+    device_str: String representation of the current device.
+    step_num: Integer. Current step number.
+    loss: Float. Current loss.
+    rate: Float. The examples/sec rate for the current batch.
+    global_rate: Float. The average examples/sec rate since training began.
+  """
+  print('[{}]({}) Loss={:.5f} Rate={:.2f}, GlobalRate={:.2f}, Time={}'.format(
```

Comment:
nit: Can we make the delimiters consistent (I see both ` ` and `, `. I think ` ` would be fine?).

Reply:
Space without comma looks fine according to our current regexes: https://cs.corp.google.com/piper///depot/google3/cloud/cluster/testing/tpu/tputest.py?rcl=266961178&l=147

Updated